expo/gbe/views/create_volunteer_view.py

```diff
@@ -19,31 +26,65 @@
     validate_profile,
 )
 from gbetext import (
-    default_volunteer_submit_msg
+    default_volunteer_submit_msg,
+    default_volunteer_no_interest_msg,
+    default_volunteer_no_bid_msg
 )
 
 
+def no_vol_bidding(request):
+    user_message = UserMessage.objects.get_or_create(
+                    view='CreateVolunteerView',
+                    code="NO_BIDDING_ALLOWED",
+                    defaults={
+                        'summary': "Volunteer Bidding Blocked",
+                        'description': default_volunteer_no_bid_msg})
+    messages.error(request, user_message[0].description)
+    return HttpResponseRedirect(reverse('home', urlconf='gbe.urls'))
+
+
 @login_required
 @log_func
 def CreateVolunteerView(request):
     page_title = 'Volunteer'
     view_title = "Volunteer at the Expo"
     profile = validate_profile(request, require=False)
+    formset = []
     if not profile:
         return HttpResponseRedirect(reverse('profile',
                                             urlconf='gbe.urls') +
                                     '?next=' +
                                     reverse('volunteer_create',
                                             urlconf='gbe.urls'))
+    try:
+        conference = Conference.objects.filter(accepting_bids=True).first()
+        windows = conference.windows()
+        available_interests = AvailableInterest.objects.filter(
+            visible=True).order_by('interest')
+    except:
+        return no_vol_bidding(request)
+    if len(windows) == 0 or len(available_interests) == 0:
+        return no_vol_bidding(request)
     if request.method == 'POST':
         form = VolunteerBidForm(
             request.POST,
-            available_windows=Conference.current_conf().windows(),
-            unavailable_windows=Conference.current_conf().windows())
-        if form.is_valid():
+            available_windows=windows,
+            unavailable_windows=windows)
+        valid_interests = True
+        like_one_thing = False
+        for interest in available_interests:
+            interest_form = VolunteerInterestForm(
+                request.POST,
+                initial={'interest': interest},
+                prefix=str(interest.pk))
+            formset += [interest_form]
+            if interest_form.is_valid():
+                if int(interest_form.cleaned_data.get('rank')) > 3:
+                    like_one_thing = True
+            else:
+                valid_interests = False
```

Comment:
Worst case, the three loops do the same work as the one loop, so nothing is lost. More likely, python is able to optimize the loops in the filter and the any, so we save some work. In either case, the code is infinitely more readable which is always the first thing to shoot for. (Knuth's law)
Feel free to use an `all` instead of an `any` - this was a quick example, not drop-in code.